### ISSUE TITLE
userUnreadMessages gives 'Undefined offset: -1' 

### ISSUE THREAD

**Issue by bstrx** [cmgmyr/laravel-messenger#159]
Hello, I get errors when using userUnreadMessages having 0 messages:
```
ErrorException: Exception 'ErrorException' with message 'Undefined offset: -1' in /var/www/project/vendor/laravel/framework/src/Illuminate/Support/Collection.php:1187

Sample stack trace
…/var/www/project/vendor/laravel/framework/src/Illuminate/Support/
Collection.php (1187)
…/var/www/project/vendor/cmgmyr/messenger/src/Cmgmyr/Messenger/Models/
Thread.php (369)
…ages called at /var/www/project/app/Http/Controllers/Api/
MessagesController.php (32)
```

in this:
```php
        $i = count($messages) - 1;
        while ($i) {
            if ($messages[$i]->updated_at->gt($participant->last_read)) {
                array_push($unread, $messages[$i]);
            } else {
                break;
            }
            --$i;
        }
```

**Comment by cmgmyr** [comment]
Hey, thanks for reporting this. What versions of Laravel and the Messenger package are you working with?

**Comment by bstrx** [comment]
@cmgmyr 
2.12 Messenger and v5.2.45 laravel

**Comment by cmgmyr** [comment]
@bstrx unfortunately, I wasn't able to duplicate your issue locally, but I did make a few changes that might have fixed it for you. Can you update your composer.json file to use `dev-master` and see if this fixes it?

**Comment by bstrx** [comment]
@cmgmyr thanks! I will monitor dev version during the day and give a feedback

**Comment by bstrx** [comment]
@cmgmyr it seems to be working nicely!)

**Comment by cmgmyr** [comment]
@bstrx awesome! I'll be officially tagging this shortly - I'll update this issue once I do it, but I'm going to close it for now.

**Comment by cmgmyr** [comment]
this has now been released in `2.13`
